packages/fluentui/docs/src/prototypes/FormValidation/FormHooks.tsx

```diff
@@ -0,0 +1,62 @@
+import * as React from 'react';
+import { Form } from '@fluentui/react-northstar';
+import { useForm } from 'react-hook-form';
+
+const FormValidateHooks = () => {
+  const { register, handleSubmit, errors, setValue, watch, triggerValidation } = useForm({
+    mode: 'onBlur',
+  });
+
+  const onSubmit = data => {
+    console.log('errors: ', errors);
+    console.log('data: ', data);
+  };
+
+  const firstNameWatched = watch('firstName');
+
+  React.useEffect(() => {
+    register(
+      { name: 'city' },
+      {
+        required: { value: true, message: 'city is required' },
+      },
+    );
+  }, [register]);
+
+  const handleMultiChange = selectedOption => {
+    setValue('city', selectedOption);
+  };
+  return (
+    <Form onSubmit={handleSubmit(onSubmit)}>
+      <Form.Input
+        label="First Name"
+        id="first-name-hooks"
+        required
+        name="firstName"
+        value={firstNameWatched}
+        inputRef={register({
+          required: { value: true, message: 'first name is required' },
+          validate: val => (val.length <= 4 ? 'Too Short' : null),
+        })}
+        errorMessage={errors.firstName?.message}
+        showSuccessIndicator={!errors.firstName && firstNameWatched?.length >= 4}
```

Comment:
I would like to avoid dependency on `firstNameWatched` in examples and use `dirty` instead:

```suggestion
        showSuccessIndicator={dirty && !errors.firstName}
```